src/compiler/transformers/ts.ts

```diff
@@ -2223,6 +2216,10 @@ namespace ts {
                 /*location*/ node
             );
 
+            // Keep modifiers in case of async functions

```

Comment:
We should consider inserting `modifiers` as the first parameter of `createFunctionExpresson`. Generally we should avoid mutating a node after we call the factory method. This also visits modifiers out-of-order with respect to other child nodes of the function.